server/src/main/java/com/github/sgov/server/controller/WorkspaceController.java

```diff
@@ -1,60 +1,172 @@
 package com.github.sgov.server.controller;
 
-import com.github.sgov.server.dao.WorkspaceDao;
+import com.github.sgov.server.model.VocabularyContext;
+import com.github.sgov.server.model.Workspace;
 import com.github.sgov.server.service.IdentifierResolver;
+import com.github.sgov.server.service.WorkspaceService;
+import com.github.sgov.server.util.Constants.QueryParams;
+import com.github.sgov.server.util.Vocabulary;
+import cz.cvut.kbss.jsonld.JsonLd;
 import io.swagger.annotations.Api;
 import io.swagger.annotations.ApiImplicitParam;
 import io.swagger.annotations.ApiOperation;
 import io.swagger.annotations.ApiParam;
-import java.util.List;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
 import org.springframework.util.MimeTypeUtils;
-import org.springframework.web.bind.annotation.GetMapping;
-import org.springframework.web.bind.annotation.RequestMapping;
-import org.springframework.web.bind.annotation.RequestParam;
-import org.springframework.web.bind.annotation.ResponseBody;
-import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.bind.annotation.*;
 import org.topbraid.shacl.validation.ValidationReport;
 
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+
 @RestController
-@RequestMapping("/workspace")
+@RequestMapping("/workspaces")
 @Api(tags = "Workspace")
 @SuppressWarnings("checkstyle:MissingJavadocType")
 public class WorkspaceController extends BaseController {
 
-    private final WorkspaceDao workspaceDao;
+    private static final Logger LOG = LoggerFactory.getLogger(WorkspaceController.class);
+    private final WorkspaceService workspaceService;
 
     @Autowired
-    public WorkspaceController(WorkspaceDao workspaceDao, IdentifierResolver idResolver) {
+    public WorkspaceController(WorkspaceService workspaceService, IdentifierResolver idResolver) {
         super(idResolver);
-        this.workspaceDao = workspaceDao;
+        this.workspaceService = workspaceService;
+    }
+
+    @GetMapping(produces = {
+            MediaType.APPLICATION_JSON_VALUE, JsonLd.MEDIA_TYPE})
+    @ApiOperation(value = "Retrieve all workspaces.")
+    @ResponseBody
+    public List<Workspace> getAllWorkspaces() {
+        return workspaceService.findAll();
+    }
+
+    @GetMapping(value = "/current", produces = {
+            MediaType.APPLICATION_JSON_VALUE, JsonLd.MEDIA_TYPE})
+    @ApiOperation(value = "Retrieve current workspace of authenticated user.")
+    @ResponseBody
+    public Workspace getCurrentWorkspace() {
+        return workspaceService.getCurrentWorkspace();
+    }
+
+    @DeleteMapping(value = "/current")
```

Comment:
This is to logout from workspace?

Reply:
- Good point, it is very unintuitive. (Why DELETE: I am trying to delete relator between current user and selected workspace)


It was like this:
GET /workspaces/current
PUT /workspaces/{fragment}/current
DELETE /workspaces/current


I fixed it like this (although not perfect but hopefully more intuitive):
GET /users/current/workspace
PUT /users/current/workspace/{fragment}
DELETE /users/current/workspace